compiler/src/parsing/lexer.mll

```diff
@@ -218,7 +224,8 @@ and read_str buf =
   | "\\\"" { Buffer.add_char buf '"'; read_str buf lexbuf }
   | "\\\\" { Buffer.add_char buf '\\'; read_str buf lexbuf }
   | num_esc { add_code_point buf (lexeme lexbuf) (lexbuf_loc lexbuf); read_str buf lexbuf }
-  | [^ '"' '\\']+ { process_newlines lexbuf; Buffer.add_string buf (lexeme lexbuf); read_str buf lexbuf }
+  | '\n' as char { Buffer.add_char buf char; new_line lexbuf; read_str buf lexbuf }
+  | [^ '"' '\\' '\n']+ { Buffer.add_string buf (lexeme lexbuf); read_str buf lexbuf }
```

Comment:
Just trying to learn the lexer.. Is specifying \n here just for safety, as it should have been captured by the line above, or necessary?

Reply:
@marcusroberts The `^` at the start of the character class negates it, so this class includes things which are _not_ `\n`.

Reply:
(Thinking about it, though, you're right; I suppose it is redundant)

Reply:
It's necessary in this case! Maybe slightly more regex reasons rather than lexing reasons, but the `+` is greedy and will keep consuming until it hits a `"` or `\`, so if this rule is in effect (without the `\n` in there), it'll capture all of the newlines without stopping to let us process them.